### PR TITLE
feature(pu): add mujoco, minigrid and bsuite related envs/configs, add muzero_recurrent related mlp/conv model and related configs

### PR DESCRIPTION
**Pull request by puyuan1996**
- add mujoco, minigrid and bsuite related envs and configs
- add muzero_recurrent related mlp/conv model and related configs
- add random_collect_episode_num, eps_greedy, rnd reward model option to tackle sparse reward problem
- add manually augment extreme action option in sampled efficientzero + mujoco

### CODE REVIEW

File: lzero/policy/muzero.py [hunk 1]
```diff
@@ -372,6 +395,10 @@ def _forward_learn(self, data: Tuple[torch.Tensor]) -> Dict[str, Union[float, in
             # ==============================================================
             policy_loss += cross_entropy_loss(policy_logits, target_policy[:, step_i + 1])
 
+            prob = torch.softmax(policy_logits, dim=-1)
```
Comment (PaParaZz1) [inline lzero/policy/muzero.py:398]: why there is no switch to control it

File: lzero/model/muzero_model_mlp.py [hunk 2]
```diff
@@ -432,8 +419,8 @@ def forward(self, state_action_encoding: torch.Tensor) -> Tuple[torch.Tensor, to
             next_latent_state = x + latent_state
             next_latent_state_encoding = self.fc_dynamics_2(next_latent_state)
         else:
-            next_latent_state = self.fc_dynamics(state_action_encoding)
-            next_latent_state_encoding = next_latent_state
+            next_latent_state = self.fc_dynamics_1(state_action_encoding)
```
Comment (PaParaZz1) [inline lzero/model/muzero_model_mlp.py:422]: if other models have the same cases, you need to modify all of them

File: lzero/entry/utils.py [hunk 3]
```diff
@@ -1,9 +1,47 @@
...
+    assert policy_cfg.random_collect_episode_num > 0
+
+    random_policy = RandomPolicy(cfg=policy_cfg)
+    # set the policy to random policy
+    collector.reset_policy(random_policy.collect_mode)
+
+    collect_kwargs = {}
+    # set temperature for visit count distributions according to the train_iter,
+    # please refer to Appendix D in MuZero paper for details.
+    collect_kwargs['temperature'] = 1
```
Comment (PaParaZz1) [inline lzero/entry/utils.py:29]: we need to remove this in random collect

File: zoo/classic_control/cartpole/config/cartpole_sampled_efficientzero_config.py [hunk 4]
```diff
@@ -41,6 +41,7 @@
             discrete_action_encoding_type='one_hot',
             norm_type='BN', 
         ),
+        # policy_entropy_loss_weight=0.005,
```
Comment (PaParaZz1) [inline zoo/classic_control/cartpole/config/cartpole_sampled_efficientzero_config.py:44]: remove this
